source/common/quic/envoy_quic_utils.cc

```diff
@@ -17,14 +17,14 @@ namespace Quic {
 // the heap allocation is too expensive.
 Network::Address::InstanceConstSharedPtr
 quicAddressToEnvoyAddressInstance(const quic::QuicSocketAddress& quic_address) {
-  return quic_address.IsInitialized()
-             ? Network::Address::addressFromSockAddr(quic_address.generic_address(),
-                                                     quic_address.host().address_family() ==
-                                                             quic::IpAddressFamily::IP_V4
-                                                         ? sizeof(sockaddr_in)
-                                                         : sizeof(sockaddr_in6),
-                                                     false)
-             : nullptr;
+  if (!quic_address.IsInitialized()) {
+    return nullptr;
+  }
```

Comment:
need to know you are on a worker thread here.

Reply:
makes sense.